### PR TITLE
Add Knative GKE demo

### PR DESCRIPTION
**Pull request by lblackstone**

### CODE REVIEW

File: gcp-ts-gke-knative/README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# gcp-ts-gke-knative
+All-in-one demo of Knative on GKE
+
+**Not intended for production use**
```
Comment (lukehoban) [inline gcp-ts-gke-knative/README.md]: Why is this needed?  Should we at least qualify this statement?  Is it because of Pulumi - or because of Knative?  If needed, could we soften the language?

File: gcp-ts-gke-knative/README.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+This demo creates a 3-node GKE cluster with `n1-standard-4` instances.
+Remember to tear down these resources after use to avoid unwanted charges.
+
+## Prerequisites
+
+* GCP account configured with the `Kubernetes Engine Cluster Admin` role
+
+## Known issues:
+
+1. Stack operations get very slow with so many resources in one stack.
```
Comment (lukehoban) [inline gcp-ts-gke-knative/README.md]: I think we probably want to fix this (or fix the product!) before publishing the example.  This doesn't feel like it's putting a good experience in front of users.

File: gcp-ts-gke-knative/README.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+Splitting the deployment into  multiple stacks (e.g., GKE cluster,
+Istio/Knative, app) solves this problem, and is more representative
+of production use.
+
+1. `pulumi destroy` fails with an error:
```
Comment (lukehoban) [inline gcp-ts-gke-knative/README.md]: Can we fix this in the product or example before publishing this?  It seems a key goal we have here is to take this burden off of users.

Comment (lblackstone) [reply gcp-ts-gke-knative/README.md]: Yes, I haven't had time to dig into the root cause yet, but we should fix this in the k8s provider.

File: gcp-ts-gke-knative/README.md (hunk 2 above)
Comment (lblackstone) [reply gcp-ts-gke-knative/README.md]: +1 on fixing the product. I wasn't sure if it was worth the effort to split up in stacks for this demo.

File: gcp-ts-gke-knative/README.md (hunk 1 above)
Comment (lblackstone) [reply gcp-ts-gke-knative/README.md]: A little bit of both: 
* Knative + Istio are complex and we haven't tested this at all beyond the "hello world" demo.
* The current AIO stack is almost unusably slow.

**Comment by justinvp** [comment]
@lblackstone, what's the status of this PR?

**Comment by lblackstone** [comment]
I haven't had a chance to come back to it, and likely won't any time soon. I think it's pretty heavy for an example anyway, so I'll close the PR and we can resurrect it if we want later.
